dvc/output.py

```diff
@@ -849,7 +849,8 @@ def get_dir_cache(self, **kwargs):
         try:
             objects.check(self.odb, obj)
         except FileNotFoundError:
-            remote = self.repo.cloud.get_remote_odb(self.remote)
+            if self.remote:
```

Comment:
This is getting irritating, will need to look into getting rid of `get_dir_cache` soon. :disappointed:

Reply:
Shouldn't `-r <remote>` be of higher priority?

Reply:
My second priority would be to skip not specified remotes.

Reply:
Though, for `dir` cache, I would not care that much. 🤷🏼

Reply:
`remote: ` has higher prioirity than `-r`, as documented in https://github.com/iterative/dvc.org/pull/2761/files . Similar how `dvc import`ed remotes are not overriden by `-r`